caravel/assets/javascripts/explore.js

```diff
@@ -400,6 +383,78 @@ function initExploreView() {
     });
     $(this).remove();
   });
+
+  function prepSaveDialog() {
+    var setButtonsState = function () {
+      var add_to_dash = $("input[name=add_to_dash]:checked").val();
+      if (add_to_dash === 'existing' || add_to_dash === 'new') {
+        $('.gotodash').removeAttr('disabled');
+      } else {
+        $('.gotodash').prop('disabled', true);
+      }
+    };
+    var url = '/dashboardmodelviewasync/api/read';
+    url += '?_flt_0_owners=' + $('#userid').val();
+    $.get(url, function (data) {
+      var choices = [];
+      for (var i=0; i< data.pks.length; i++) {
+        choices.push({ id: data.pks[i], text: data.result[i].dashboard_title });
+      }
+      $('#save_to_dashboard_id').select2({
+        data: choices,
+        dropdownAutoWidth: true
+      }).on("select2-selecting", function () {
+        $("#add_to_dash_existing").prop("checked", true);
+        setButtonsState();
+      });
+    });
+
+    $("input[name=add_to_dash]").change(setButtonsState);
+    $("input[name='new_dashboard_name']").on('focus', function () {
+      $("#add_to_new_dash").prop("checked", true);
+      setButtonsState();
+    });
+    $("input[name='new_slice_name']").on('focus', function () {
+      $("#save_as_new").prop("checked", true);
+      setButtonsState();
+    });
+    $('#btn_modal_save').click(function () {
```

Comment:
this is a pretty good explanation: http://stackoverflow.com/questions/9122078/difference-between-onclick-vs-click

basically `$el.on('click', function() {})` is more performant because one handler is created rather than a handler for every element. this would matter more for cases like binding clicks for multiple elements, like `$('.btn')`. in this case it's pretty safe, but i think it's generally a good pattern to follow.  it also allows you to unbind the handler by using `$el.off('click', function() {});`

it would be a great idea to change them all! 🎉 

as we move towards react we won't have to use this function as much. (if at all)